### PR TITLE
Refactor and test split message

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

While debugging some stuff, I came across this potential refactor. It's completely optional and functions identically to the previous implementation. Even if we don't want to keep the refactor, we can keep the new test.

Closes: #XXX

### CODE REVIEW

File: types/shares_test.go [hunk 1]
```diff
@@ -456,6 +456,41 @@ func TestContigShareWriter(t *testing.T) {
 	assert.Equal(t, txs, resTxs)
 }
 
+func TestSplitMessage(t *testing.T) {
```
Comment (evan-forbes) [inline types/shares_test.go:459]: while this test is a nice to have, most of the testing is done by the round trip tests https://github.com/celestiaorg/celestia-core/blob/843c996acd2dc544db68f9fa870d4e7be3f3af63/types/shares_test.go#L218

File: types/share_splitting.go [hunk 2]
```diff
@@ -61,43 +61,32 @@ func (msw *MessageShareWriter) Count() int {
...
-	shares = append(shares, NamespacedShare{firstRawShare, nid})
-	rawData = rawData[consts.MsgShareSize:]
-	for len(rawData) > 0 {
-		shareSizeOrLen := min(consts.MsgShareSize, len(rawData))
+	dataSize := len(rawData)
+	count := (dataSize / consts.MsgShareSize)
+	if dataSize%consts.MsgShareSize != 0 {
+		count++
+	}
+	shares := make([]NamespacedShare, count)
```
Comment (evan-forbes) [inline types/share_splitting.go:74]: the main optimization is simply creating the slice ahead of time. probably super small, but hey :golf:

File: types/share_splitting.go [hunk 3]
```diff
@@ -61,43 +61,32 @@ func (msw *MessageShareWriter) Count() int {
...
-		shares = append(shares, share)
-	} else { // len(rawData) > MsgShareSize
-		shares = append(shares, splitMessage(rawData, nid)...)
-	}
+	shares = append(shares, splitMessage(rawData, nid)...)
 	return shares
 }
 
-// splitMessage breaks the data in a message into the minimum number of
-// namespaced shares
```
Comment (adlerjohn) [inline types/share_splitting.go:80]: Was there a reason to remove documentation?

File: types/share_splitting.go [hunk 4]
```diff
@@ -61,43 +61,32 @@ func (msw *MessageShareWriter) Count() int {
...
-		rawData[:consts.MsgShareSize]...,
-	)
-	shares = append(shares, NamespacedShare{firstRawShare, nid})
-	rawData = rawData[consts.MsgShareSize:]
-	for len(rawData) > 0 {
-		shareSizeOrLen := min(consts.MsgShareSize, len(rawData))
+	dataSize := len(rawData)
+	count := (dataSize / consts.MsgShareSize)
+	if dataSize%consts.MsgShareSize != 0 {
+		count++
```
Comment (adlerjohn) [inline types/share_splitting.go:72]: Document the logic here for why you need to increment

Comment (evan-forbes) [reply types/share_splitting.go:76]: good idea 1b766ee041186d6e3197caee2403ee5538f5df94

File: types/share_splitting.go (hunk 3 above)
Comment (evan-forbes) [reply types/share_splitting.go]: :grimacing: whoops, good catch added back in 1b766ee041186d6e3197caee2403ee5538f5df94

**Comment by evan-forbes** [comment]
closing this as its super outdated and non-critical. its also unclear of the benefit as this isn't benchmarked

**Comment by evan-forbes** [comment]
the test wouldn't hurt tho

**Pull request closed without merging**